persistent-mysql/Database/Persist/MySQL.hs

```diff
@@ -378,15 +378,20 @@ findMaxLenOfColumn allDefs name col =
            maxLenAttr <- find ((T.isPrefixOf "maxlen=") . T.toLower) (fieldAttrs fieldDef)
            readMaybe . T.unpack . T.drop 7 $ maxLenAttr
 
--- | Helper for 'AddRefence' that finds out the 'entityId'.
+dbIdColumns2 :: EntityDef -> [DBName]
+dbIdColumns2 t = case entityPrimary t of
+    Just pdef -> map fieldDB (compositeFields pdef)
+    Nothing   -> [fieldDB (entityId t)]
```

Comment:
@gregwebs I like the `entityKeyFields` function you added in #464 better than this one. Thinking I'll just wait until you merge than, then use `map fieldDB $ entityKeyFields entDef` in my own code.